src/App/Cart/Service/CartCalculationService.php

```diff
@@ -23,13 +24,15 @@ public function calculateAllowedPackageTypes(PdkCart $cart): PackageTypeCollecti
         return PackageTypeCollection::fromAll()
             ->sortBySize(true)
             ->filter(function (PackageType $packageType) use ($cart) {
-                $lineHasPackageType = $cart->lines->containsStrict('product.settings.packageType', $packageType->name);
+                $packageTypeName    = $packageType->name;
+                $lineHasPackageType = $cart->lines->containsStrict('product.settings.packageType', $packageTypeName);
 
-                if ($lineHasPackageType && DeliveryOptions::PACKAGE_TYPE_MAILBOX_NAME === $packageType->name) {
-                    return $this->calculateMailboxPercentage($cart) <= 100.0;
+                if ($lineHasPackageType && DeliveryOptions::PACKAGE_TYPE_MAILBOX_NAME === $packageTypeName) {
+                    return $this->calculateMailboxPercentage($cart) <= 100.0
+                        && $this->isWeightUnderPackageTypeLimit($cart, $packageTypeName);
```

Comment:
je hebt deze functie mooi generiek gemaakt, maar moet je die dan niet altijd uitvoeren, ook als het geen mailbox is?